themes/custom/c2d_theme/c2d_theme.theme

```diff
@@ -53,6 +54,17 @@ function c2d_theme_preprocess_page(array &$variables) {
  */
 function c2d_theme_preprocess_block(&$variables) {
 
+  $block_content = $variables['elements']['content']['#block_content'] ?? '';
+  if ($block_content instanceof BlockContentInterface) {
+    $block_type = $block_content->bundle();
+
```

Comment:
why we are limit classes to these 2 types?
what about:
$variables['attributes']['class'][] = $block_type . '-block-type';